data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -234,15 +235,18 @@ private void addExceptionHandlers() {
     app.exception(BadRequestException.class, this::badRequest);
     app.exception(JsonProcessingException.class, this::badRequest);
     app.exception(IllegalArgumentException.class, this::badRequest);
-    // Add catch-all handler
-    app.exception(
-        Exception.class,
-        (e, ctx) -> {
-          LOG.error("Failed to process request to URL {}", ctx.url(), e);
-          ctx.status(SC_INTERNAL_SERVER_ERROR);
-          setErrorBody(
-              ctx, () -> BadRequest.internalError(jsonProvider, "An unexpected error occurred"));
-        });
+    app.exception(Exception.class, this::catchAllExceptionHandler);
+  }
+
+  private void catchAllExceptionHandler(final Throwable throwable, final Context context) {
```

Comment:
Probably need to apply the same fix to the validator API too.  Splitting this out to a separate class probably makes sense as it should be the same for both I think.